src/compiler/crystal/macros/methods.cr

```diff
@@ -820,6 +820,16 @@ module Crystal
         else
           raise "StringLiteral#to_i: #{@value} is not an integer"
         end
+      when "to_utf16"
+        interpret_check_args do
+          slice = @value.to_utf16
+          # include the trailing zero that isn't counted in the slice but was
+          # generated by String#to_utf16 so the literal can be passed to C
+          # functions that expect a null terminated UInt16*
+          slice = Slice(UInt16).new(slice.to_unsafe, slice.size + 1)
+          args = slice.to_a { |codepoint| NumberLiteral.new(codepoint).as(ASTNode) }
+          Call.new(Path.global("Slice(UInt16)"), "literal", args)
```

Comment:
Having a `Path` that isn't actually a valid `Path` will cause problems, since the `Call` itself can be introspected. Also it should be `::Slice(::UInt16)`:

```suggestion
          Call.new(Generic.new(Path.global("Slice"), [Path.global("UInt16")] of ASTNode), "literal", args)
```

Reply:
Thanks @HertzDevil! I thought about it then got distracted :bow: